runtime/drive_handler.go

```diff
@@ -127,9 +127,9 @@ func (h *stubDriveHandler) createStubDrive(driveID, path string) error {
 
 // SetDrives will set the given drives and the offset to which the stub drives
 // start.
-func (h *stubDriveHandler) SetDrives(offset int64, d []models.Drive) {
+func (h *stubDriveHandler) SetDrives(d []models.Drive) {
 	h.drives = d
-	h.stubDriveIndex = offset
+	h.stubDriveIndex = 0
```

Comment:
nit: I think we can just remove setting `h.stubDriveIndex` now since the struct will default the value to initially be 0 anyways.

Reply:
`SetDrives()` can be called anytime during the handler's lifecycle. How about returning an error from this function, since we assume that SetDrives should be called before calling PatchStubDrive().

Another type-safe way of doing that is getting rid of SetDrives() and initializing the handler with stub drives. Making things immutable is good, while it may make this PR a bit bigger though.

Reply:
One thing to keep in mind is that this is a `main` package so I don't think any customers would be able to use the code. I get what you're saying though in that we'd be relying on our other code in this package to only call `SetDrives` once. That's technically enforced right now because you can only call CreateVM once (due to a `s.vmStartOnce` in `service.go`), but it does still all feel a bit fragile.

I'd be good with either approach you suggested. Initializing the stub drive handler w/ a container count only when the VM has started would be most in line with the rest of the code though (i.e. we currently initialize `s.agentClient`, `s.eventBridgeClient` and `s.exitAfterAllTasksDeleted` only when the VM has started and guard their initialization with `s.vmStartOnce`).